wordpress-mobile/WordPress-iOS
# Jetpack App (Emphasis): Add a new JetpackScreenshotGeneration target

Part of  #16395 

This PR adds a new JetpackScreenshotGeneration target.

## Description
- Duplicated the WordPressScreenshotGeneration target
- Added UI tests for App Store screenshot screens (Design ref: pcdRpT-mz-p2)
    - My Site
    - Activity Log
    - Scan
    - Backup
    - Stats 

## How to test

Note: My plan is to update WordPressMocks in a separate PR, hence disabling stubbing in the testing directions in this PR.

1. Disable WireMock by removing it from UI Test launch arguments `XCTest+Extensions.swift:73`
```swift
app.launchArguments = ["-wpcom-api-base-url", "-no-animations", "-ui-testing"]
// app.launchArguments = ["-wpcom-api-base-url", WireMock.URL().absoluteString, "-no-animations", "-ui-testing"]
```
2. Replace credentials on `JetpackScreenshotGeneration.swift:25` with credentials for an account that has a site where Backup and Scan are enabled (i.e. Jetpack Complete test site)
3. If needed, add the following lines in `JetpackScreenshotGeneration.swift` to make sure you're switching to a site that has both Backup and Scan
```swift
// Get My Site screenshot
let mySite = MySiteScreen()
    .showSiteSwitcher()
    .switchToSite(withTitle: <siteAddress>)
    .thenTakeScreenshot(1, named: "MySite")
```
4. ⚠️ Delete any existing Jetpack apps from your simulator
5. ⚠️ Make sure to disconnect the hardware keyboard in our simulator settings (Cmd+Shift+K)... otherwise the tests will fail
4. Finally, test the `JetpackScreenshotGeneration` scheme
5. ✅ `testGenerateScreenshots` should pass


## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
